cli/src/commands/InitCommand.ts

```diff
@@ -1,9 +1,242 @@
+      emptyConfig,
+      ts.ScriptTarget.Latest
+    );
+
+    return {
+      isExistingConfig: false,
+      sourceFile,
+      importModules: ["@openapi-codegen/cli"],
+    };
+  }
+
+  private async askForFile(): Promise<FileOptions> {
+    return {
+      relativePath: await this.prompt.input({
+        message: "Relative path",
+        hint: "Example: ./openapi.json",
+      }),
+      source: "file",
+    };
+  }
+
+  private async askForUrl(): Promise<UrlOptions> {
+    return {
+      source: "url",
+      url: await this.prompt.input({
+        message: "Url",
+        hint: "Example: https://.../openapi.json",
+      }),
+    };
+  }
+
   async execute() {
-    this.context.stdout.write(`Init the config file`);
+    const userConfigPath = path.join(
+      process.cwd(),
+      this.config || "openapi-codegen.config.ts"
+    );
+
+    const config = await this.getConfigSourceFile(userConfigPath);
+
+    const source = await this.prompt.select({
+      choices: [
+        { label: "File", value: "file" as const },
+        { label: "Url", value: "url" as const },
+        // { label: "Github", value: "github" as const },
+      ],
+      message: "Select the source of your OpenAPI",
+    });
+
+    const from: FromOptions =
+      source === "file"
+        ? await this.askForFile()
+        : source === "url"
+        ? await this.askForUrl()
+        : await this.prompt.github();
+
+    const namespace = format.camel(
+      await this.prompt.input({
+        message: "What namespace do you want for your API?",
+      })
+    );
+
+    const plugin = await this.prompt.select<Plugin>({
+      message: "What do you want to generate?",
+      choices: [
+        { label: "Types only", value: "typescript/types-only" },
+        { label: "Generic Fetchers", value: "typescript/fetch" },
+        { label: "React Query components", value: "typescript/react-query" },
+      ],
+    });
+
+    const outputDir = await this.prompt.input({
+      message: "Which folder do you want to generate?",
+    });
+    this.prompt.close();
+
+    const configProperty = generateConfigProperty({
+      namespace,
+      options: {
+        from,
+        outputDir,
+        plugin,
+      },
+    });
+
+    const importsToInsert = getImports(plugin);
+
+    const updatedConfigSourceFile = updateConfig({
+      sourceFile: config.sourceFile,
+      existingImports: config.importModules,
+      importsToInsert,
+      configProperty,
+    });
+
+    const printer = ts.createPrinter({
+      newLine: ts.NewLineKind.LineFeed,
+      removeComments: false,
+    });
+
+    const prettierConfig = await prettier.resolveConfig(process.cwd());
+
+    const updatedConfig = prettier.format(
+      printer.printFile(updatedConfigSourceFile),
+      { parser: "babel-ts", ...prettierConfig }
+    );
+
+    if (this.dryRun) {
+      this.context.stdout.write(
+        highlight(updatedConfig, {
+          language: "typescript",
+        })
+      );
```

Comment:
![image](https://user-images.githubusercontent.com/1761469/158653985-b42afa7d-31af-478a-9ed4-2364aee61d7c.png)

Yes, I did add syntax highlighting for the `--dry-run` 😇